shsorot/sap-automation/Experimental to shsorot/sap-automation/scaleout-hsr  pull

shekharsorot opened the pull request:
pulling code from upstream experimental branch into downstream scaleout-hsr branch to start merging scaleout

shekharssorot2002 reviewed (commented):
confirm all merges to shsorot/sap-automation:main

SteffenBoThomsen committed:
Fixes for HANA VM ppg and /etc/hosts file generation (#511)

* resolve proximity_placement_group_id for HANA db

* ensure hosts.j2 correctly places at least a single whitespace when FQDN is longer than the 50 chars and increase the alignment of the hostname at 80 chars.

hdamecharla committed:
Final changes before release v3.10.0.0

hdamecharla committed:
pass the correct operation name

hdamecharla committed:
switch to using the newer parameters for high availability.

hdamecharla committed:
remove extra lines

hdamecharla committed:
Remove unnecessary tasks for SUSE and REDHAT in ACSS registration playbook

hdamecharla committed:
Add az login before Get Access Token tasks for ACSS registration

hdamecharla committed:
Add debug task for ACSS registration parameters

hdamecharla committed:
Update ACSS registration tasks in Ansible playbook

Commit:
Public network flag

Commit:
Merge branch 'experimental' of https://github.com/Azure/sap-automation into experimental

Commit:
Add the Azure Center for SAP solutions Service role for management role for the MSI

Commit:
Use Azure Center for SAP solutions administrator

hdamecharla committed:
Add ACSS virtual instance write check

hdamecharla committed:
Add DB2 keystore file handling tasks

hdamecharla committed:
Update keystore file handling and ACSS registration configuration

hdamecharla committed:
Fix DB2 keystore file check in ansible playbook

hdamecharla committed:
Fix condition for copying keystore files

hdamecharla committed:
Add DB2 restore tasks for encrypted and non-encrypted databases

Commit:
Remove the extra debug messages

hdamecharla committed:
Commented out database_high_availability and scs_high_availability settings

hdamecharla committed:
Add backward compatibility for checking required Database HA variables

hdamecharla committed:
Refactor playbook_04_00_00_db_install.yaml

hdamecharla committed:
Add DBLoad installation tasks and PAS installation tasks

hdamecharla committed:
Fix condition for secret existence check in ha_clusterpasswords.yaml

Commit:
If iscsi count == 0 no output

Commit:
Add private endpoint checks

Commit:
respect the end point flag

Commit:
Provide the means to control the Utility VM disk size

Commit:
PRovide the ability to deploy a Linux Utility VM and also control the OS Disk type

Commit:
Authentication for Linux utility VM

Commit:
Web App updates

Commit:
Add the ability to provide persisted extra parameters

Commit:
show consolidated list

Commit:
Parameter logic

Commit:
Concat the variables

Commit:
Testing

Commit:
Add the ability to source extra variables from the variable groups

hdamecharla committed:
Remove unnecessary TMPDIR environment variable

Commit:
Update all packages on RHEL based systems

hdamecharla committed:
Fix typo in LandscapeTemplate.txt

hdamecharla committed:
Fix cluster_group_location failed_when condition and export AZADHOME

hdamecharla committed:
Update SDAF version to 3.10.1.0

hdamecharla committed:
Remove duplicate failed_when condition in main.yaml

Commit:
Squashed commit of the following:

commit 51afc157af112a2c886dc658ce0461f016816262
Author: Harm Jan Stam <harmjan+git@harmjan.com>
Date:   Fri Dec 15 18:45:21 2023 +0100

    Bugfix kv-secrets debug and import tasks (#516)

    Because the operation variable has no default within the role its existence should be checked.

hdamecharla committed:
Update keystore file handling and ACSS registration configuration

hdamecharla committed:
Fix DB2 keystore file check in ansible playbook

Commit:
Web App updates

hdamecharla committed:
Merge remote-tracking branch 'refs/remotes/origin/experimental' into experimental

Commit:
Systemd-Based SAP Startup Framework

hdamecharla committed:
Update HANA Cluster and SCSERS profiles

Commit:
Debugging drop file

Commit:
debug the services retrieval

Commit:
don't fail if service does not exist

Commit:
Management DNS group

Commit:
Fix the output

Commit:
Add newline

Commit:
switch to lineinfile

Commit:
Moving the task to the correct place

Commit:
Merge commit '3f54e88e30d3e62312b3f703a63f71ea0ae94956' into experimental

Commit:
don't use the .d

Commit:
don't use the .d

Commit:
Don't do the validation

Commit:
add a try statement

Commit:
move from blockinfile

Commit:
Revert "Bugfix kv-secrets debug and import tasks (#516)"

This reverts commit 51afc157af112a2c886dc658ce0461f016816262.

Commit:
lint

Commit:
indentation

Commit:
fix indentation

hdamecharla committed:
idnentation

hdamecharla committed:
Revert "idnentation"

This reverts commit e9b61e3455d25fb9fe5766c37a2f88ed9901e404.

hdamecharla committed:
merge changes

hdamecharla committed:
Refactor HANA Cluster configuration for systemd-based SAP Startup Framework

Commit:
Merge commit 'e163e2b645c478bccb600bef749ac266009658b4' into experimental

hdamecharla committed:
Add Azure DevOps Agent configuration script to configure the agent directory and set environment variables for SAP automation deployment in the configure deployer script.

hdamecharla committed:
Update ansible_core_version to ansible_version

Commit:
clean up

Commit:
Fix the search path

hdamecharla committed:
Add Systemd-Based SAP Startup Framework for SLES

Commit:
Add the fqdn for the hostname

hdamecharla committed:
Disable and stop services, set cluster out of maintenance mode

Commit:
Attempt to fix the conditional

Commit:
Merge branch 'experimental' of https://github.com/Azure/sap-automation into experimental

Commit:
remove jinja from asserts

Commit:
Change asserts

Commit:
Ansible 2.15 support

Commit:
Fix variable name

hdamecharla committed:
Update systemd configuration for 5.6 SCSERS Pacemaker

Commit:
Simplify AVSet logic

Commit:
Merge branch 'experimental' of https://github.com/Azure/sap-automation into experimental

Commit:
avset logic update

Commit:
PPG count

hdamecharla committed:
Disable and Stop services if they exist in 5.6 SCSERS Pacemaker role

hdamecharla committed:
remove extra line in tasks

hdamecharla committed:
Add DB load balancer port check for PAS installation

hdamecharla committed:
additional debug

hdamecharla committed:
Refactor pipeline script to handle extra parameters and improve logging

hdamecharla committed:
Refactor PAS installation tasks and add DB load balancer port check

hdamecharla committed:
Fix PAS Install failure and handle extra parameters in pipeline

hdamecharla committed:
Fix comparison operator in pipeline script

hdamecharla committed:
Add verbosity level 2 to debug message in PAS Install task

hdamecharla committed:
add clear errors to stonith and health events in pcmk enablement

Commit:
Also store the hidden config file

Commit:
Merge branch 'experimental' of https://github.com/Azure/sap-automation into experimental

Commit:
Bump up Ansible to 2.15 on Ubuntu 22.04

hdamecharla committed:
Update keystore file handling and ACSS registration configuration

hdamecharla committed:
Fix DB2 keystore file check in ansible playbook

hdamecharla committed:
Update keystore file handling and ACSS registration configuration

hdamecharla committed:
Fix DB2 keystore file check in ansible playbook

Commit:
Remove the extra debug messages

Commit:
Squashed commit of the following:

commit 51afc157af112a2c886dc658ce0461f016816262
Author: Harm Jan Stam <harmjan+git@harmjan.com>
Date:   Fri Dec 15 18:45:21 2023 +0100

    Bugfix kv-secrets debug and import tasks (#516)

    Because the operation variable has no default within the role its existence should be checked.

Commit:
Systemd-Based SAP Startup Framework

hdamecharla committed:
Update HANA Cluster and SCSERS profiles

Commit:
switch to lineinfile

Commit:
Add newline

Commit:
debug the services retrieval

Commit:
Debugging drop file

Commit:
don't fail if service does not exist

Commit:
Moving the task to the correct place

Commit:
move from blockinfile

Commit:
indentation

Commit:
Don't do the validation

Commit:
Revert "Bugfix kv-secrets debug and import tasks (#516)"

This reverts commit 51afc157af112a2c886dc658ce0461f016816262.

Commit:
fix indentation

hdamecharla committed:
idnentation

Commit:
lint

hdamecharla committed:
Revert "idnentation"

This reverts commit e9b61e3455d25fb9fe5766c37a2f88ed9901e404.

hdamecharla committed:
Refactor HANA Cluster configuration for systemd-based SAP Startup Framework

hdamecharla committed:
Add Systemd-Based SAP Startup Framework for SLES

hdamecharla committed:
additional debug

hdamecharla committed:
remove extra line in tasks

hdamecharla committed:
Disable and Stop services if they exist in 5.6 SCSERS Pacemaker role

hdamecharla committed:
Add DB load balancer port check for PAS installation

hdamecharla committed:
Refactor PAS installation tasks and add DB load balancer port check

hdamecharla committed:
Merge branch 'experimental' of https://github.com/Azure/sap-automation into experimental

hdamecharla committed:
resolve merge issues from rebase

Commit:
List the resources

shekharsorot committed:
Initial commit

devanshjainms committed:
Fix iSCSI service tier typo and update iSCSI initiator name (#522)

KimForss committed:
Add support for controlling DNS A record creation for secondary names (#523)

* Add support for controlling DNS A record creation for secondary names
--------
Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

KimForss committed:
Refactor SAP transport and installation path handling (#524)

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

KimForss committed:
Use afs for shared storage (#525)

* Add support for using AFS for shared media

* Refactor storage account count logic

---------

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

KimForss committed:
Web dispatcher update (#526)

* Remove unnecessary 'become' option from playbook

* Update SAP log file paths

* Update file_path in playbook_05_04_sap_web_install.yaml

---------

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

Commit:
Fix value assignment in outputs.tf

Commit:
Add use_AFS_for_installation_media and dns_a_records_for_secondary_names properties

hdamecharla committed:
Update transform.tf to use iscsi subnet NSG if defined

Commit:
Fix typo in variable name for iSCSI NSG

Commit:
Add SAP HANA backup folder creation task

Commit:
Refactor dbload task condition in main.yaml

Commit:
Update db_lb_virtual_host in PAS and APP Install tasks

KimForss committed:
Add SAP OS configuration playbook for High Availability services (#528)

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

KimForss committed:
Provide a way to perform all deployments using a User Assigned Identity (#530)

* TF updates to enable deployments based on MSI.
* Ansible updates to properly use the custom_ names during HA deployments.
* Updates to webapp to reflect the TF updates for MSI based deployments.
---------

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

Commit:
Remove custom app_virtual_hostname variable in main.yaml

Commit:
Fix ARM_CLIENT_SECRET variable in installer.sh

Commit:
Remove the login from the script

Commit:
Add IP rules for sapmnt storage account

Commit:
Add IP rules to storage accounts

Commit:
Refactor storage account IP rules

Commit:
Fix variable interpolation in deploy script

Commit:
Add condition to check if "Use WebApp" is selected

Commit:
Fix syntax error in storage_accounts.tf

Commit:
Add network rules for sapmnt storage account

Commit:
Update deploy control plane pipeline and sap_deployer module

This commit updates the deploy control plane pipeline and sap_deployer module. Changes include handling undefined variables, exporting environment variables, and modifying conditional statements.

Commit:
Add Agent IP to storage account network rules

Commit:
Update second task

Commit:
Refactor conditional statements in module.tf

Commit:
Commented out unused code for APP_REGISTRATION_APP_ID and WEB_APP_CLIENT_SECRET

Commit:
Update USE_WEBAPP parameter to lowercase

Commit:
Update TF_VAR_use_webapp to use lower case parameter value

Commit:
Fix BOM upload and download tasks

Commit:
Update user authentication logic

Commit:
Add deployment variable to tfvar_variables.tf

Commit:
Remove unused variable "deployment"

Commit:
Refactor deployment script to support both Service Principal and Managed Identity

KimForss committed:
Provide a way to control SAS token access to storage accounts (#531)

* Add support for deploying storage accounts that use RBAC

* Add storage_use_azuread flag to azurerm provider

* Update providers.tf to conditionally set use_msi based on var.use_spn

* Add flags to disable SAS tokens for SAP Library and SAP Landscape

* Fix variable checks in deploy control plane pipeline

* testing

* Fix variable comparison in deploy script

* Refactor deployment pipeline: Remove unnecessary code and comments

* Update Azure login credentials

* Update Azure login credentials

* Remove unnecessary use_azuread_auth flag

* Update storage_use_azuread based on shared_access_key_enabled variable

* Add default OAuth authentication for storage accounts

* Only set flag for the other accounts

* Update tfvar_variables.tf: Added 'deployment' variable

* Remove unused variable "deployment"

* Add shared access key enabled variable to tfvar_variables.tf

* Remove unnecessary code in imports.tf

* Update azuread provider configuration

* Update backend configuration in sap_library backend.tf

* Remove use_msi variable from backend.tf

* Add Azure AD authentication to azurerm backend

* Add shared access key configuration for storage account

* Add Azure AD authentication to backend.tf and imports.tf for SAP Workload zone

* Update storage_use_azuread to true in providers.tf

* Add Azure AD credentials to azurerm and azuread providers

* Remove unnecessary provider configuration

* Add Azure AD authentication to azurerm backend and update azuread provider configuration

* Add use_msi and use_azuread_auth options to terraform_remote_state

* Support sas-token less access

* Add installation media storage account check and extract SAS token

* Remove unnecessary line in main.yaml

* Update account_name parameter in main.yaml

* Refactor installation media storage details tasks

* Refactor storage details and add parameter review debug

* Fix indentation in allowSharedKeyAccess condition

* Refactor validation check for allowSharedKeyAccess variable

* Add --auth-mode login

* Update BOM upload command to include SAS keys

* Update BOM Validator tasks and pre-checks

* Check the file using CLI

* Refactor bom_download.yaml task

* Fix storage account check in bom_download.yaml

* Refactor bom_download.yaml and bom_validator.yaml

* Add account information retrieval and debug task

* Allow Ansible to run as MSI

* Remove ARM_TENANT_ID from deployment scripts

* Remove unnecessary Azure login step in pipeline

* Update container name in bom_download.yaml

* Fix BOM download and iSCSI key vault dependencies

* Fix storage account firewall subnet configuration

* Add useSAS variable to install_workloadzone.sh script

* Refactor network rule addition in install_workloadzone.sh

* Update authentication mode for Azure storage blob upload

* Fix condition in bom_download.yaml

* Fix Azure subscription issue in deployment pipeline

* Update Azure subscription ID in run-ansible.yaml

* Update run-ansible.yaml to source deploy_server.sh instead of using az login with identity

* Add SAP OS configuration playbook: Set sapos storage account facts

* Remove unused variable sapbits_sas_token

* Move sas token to other task

* Always enable SAS on SAP Library storage account

* Refactor key_vault.tf to remove unnecessary condition

* Refactor set_fact task in playbook_03_bom_processing.yaml

* Fix app registration app ID formatting

* Fix app registration app ID formatting in control plane deployment pipeline

* Update bom_download.yaml

lint fixes for additional new lines

* Update pre_checks.yaml

fix additional new lines lint errors

---------

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>
Co-authored-by: hdamecharla <71097261+hdamecharla@users.noreply.github.com>

Commit:
Update DEPLOYER_RANDOM_ID_SEED in deploy control plane pipeline

Commit:
Update ARM_SUBSCRIPTION_ID usage in deploy script

hjstam committed:
Several DB2 HADR improvements (#532)

* Bugfix sub_iscsi_nsg_arm_id variable

Error: A local value with the name "sub_iscsi" has not been declared

* Add boolean to configure pacemaker for Azure scheduled events

By default the boolean is set to true.

* Add custom pacemaker NFS filesystem and SAP monitor timeouts

When using a custom NFS solution different time-outs may be required.

Refactored the cluster-Suse tasks so that it utilizes the calculated filesystem timeout and
SAP resource monitor timeout from the 5.6.1-set_runtime_facts task

* Use virtual_host names for DB2 HADR local host name configuration

* Remove tags from ignore_changes for Linux dbserver

* Add SSL certificate generation and distribution for virtual_hostname

* Encrypt communication between HADR Primary and Standby Instances

Commit:
Refactor deployment pipeline and control plane script

Commit:
Fix Azure login and set ARM_USE_MSI to true

Commit:
Add console output for using SPN and MSI

KimForss committed:
Deployer msi tests (#533)

* Add cyan color to shell variables

* Update variable retrieval in deploy control plane pipeline

* Refactor deploy_controlplane.sh script

* Remove shared_access_key_enabled variable

* Refactor variable handling in deploy control plane pipeline

* Add Azure account list command to deploy_controlplane.sh script

* Add check for subscription access

* Fix ARM_SUBSCRIPTION_ID assignment in deploy control plane pipeline

---------

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

Commit:
Update backend and provider configurations

Commit:
Update Azure AD authentication in backend.tf files

Commit:
Remove use_azuread_auth flag from terraform_remote_state

Commit:
Remove use_azuread_auth from backend and imports.tf files

Commit:
Refactor bom_download.yaml to simplify when conditions

Commit:
Update scs_server_count property to allow null values

Commit:
Update database high availability configuration

Commit:
Update ANF volume count based on database server count

Commit:
Update ANF volume count based on database server count

Commit:
feng shui

shekharsorot committed:
Feature : HANA Scale out with worker/stand by node using shared ANF storage (#536)

* Add files via upload

* Test run for debugging HANA Scale out deployment

* Test run for debugging HANA Scale out deployment

* Test run for debugging HANA Scale out deployment-2

* Test run for debugging HANA Scale out deployment-3

* Test run for debugging HANA Scale out deployment-4

* add correct ANF permissions for scale out - ANF scenario

* Merge changes for Scale out ANF into playbook_04_00_db_install for seamless integration

* Suppress root password display and mark to verbose

* temporary changes to scale-out mounting

* just role name

* add probe threshold to scs and hana db only.

* Add condition to skip mounting for database scale-out

* Update database server count condition in vm-hdb.tf

* Update ANF volume count based on database server count

* Commented out the mountpoint comparision lines in ANF mounts scaleout tasks for testing

* Commented out comparision of the log mount and numbers of hosts for scale out without HSR/pacemaker for testing

* Commented out HANA Scale-Out mounts in db_install playbook

* Update HANA installation configuration

Add listen_interface and internal_network values to HANA_2_00_install_scaleout_anf.rsp template.

Set restrict_max_mem to 'n' in HANA_2_00_install_scaleout_anf.rsp template.

Refactor playbook_04_00_00_db_install.yaml to simplify conditionals and improve readability.

* Fix ANF Mounts for Scale Out

* Update async_write_submit_active value in main.yaml

* debug subnet prefix from instance metadata

* Refactor database configuration and installation tasks

* Fix SAP system start and stop commands

* Update environment variables and paths for HDB installation and SAP mounts

---------

Co-authored-by: hdamecharla <hdamecharla@microsoft.com>

KimForss committed:
Support for HANA scaleout for ANF volumes (#534)

* Update database server count and ANF volumes to support scale out scenarios

* Update hana_shared output in outputs.tf

* Update db_server_count to include stand_by_node_count

* Add output for disks attached to virtual machines

* Update HANA Data volumes description and value

* Fix keyvault check in deploy_controlplane.sh

* Update hana_data and hana_log paths in module.tf

* Refactor ANF volume outputs in HDB node module

* Refactor agent_pat and agent_pool variables in sap_deployer module.tf

* Refactor code and add TFVars support

* Refactor ANF_pool_settings in variables_local.tf

* Fix standby node count description and ANF volume count calculation

* Refactor data and log volume count calculations in variables_local.tf

* Fix data and log volume count calculation in variables_local.tf

* Lint errors

* Lint Fixes part 2

---------

Co-authored-by: Kimmo Forss <kimforss@microsoft.com>

shekharsorot committed:
Merge branch 'Azure:experimental' into experimental

Commit:
Refactor HANA Data and Log volumes in outputs.tf

Commit:
Add oraclelinux8.9 repository and packages

Commit:
Update SELinux configuration for Oracle Linux 8

Commit:
Update SELinux state to disabled in sap-notes role

shekharsorot committed:
Merge branch 'Azure:experimental' into experimental

Commit:
Fix Oracle Data Guard setup for secondary node

Commit:
Update app_virtual_hostname variable in main.yaml

Commit:
Add Oracle Data Guard post-processing task to restart lsnrctl on Secondary

shekharsorot committed:
Merge branch 'Azure:experimental' into experimental
